### PR TITLE
Allow user to change selected port

### PR DESCRIPTION
**Pull request by yousfiSaad**
Hello,
In this PR, I tried to improve the dev user experience by implementing the following:
- Allowing the client (react client) to request for available serial ports
- Allowing the user to change the selected serial port
- On initial load, the port witch has "Arduino" in the manufacturer property is selected by default

There is no need to specify the port on launch, and you can change the port using the UI, in case you are using multiple Arduinos

Have a good day

### CODE REVIEW

**Comment by yousfiSaad** [comment]
image
image

File: react-serial-ui/server/src/index.js [hunk 1]
```diff
@@ -34,46 +34,67 @@ if (NODE_ENV === "dev") {
...
 io.on("connection", (socket) => {
   const logs = [];
-  const handleUpload = getHandleUpload(logs);
-  parser.on("data", (data) => {
-    logs.push(data);
-    socket.emit("logs", logs);
-  });
+  const handleUpload = {};
+  handleUpload.current = (data) => {
+    console.log("Not yet connected");
```
Comment (acbeni) [inline react-serial-ui/server/src/index.js:75]: Shouldn't this go to logs?

File: react-serial-ui/src/components/ChangePort.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Dialog from '@material-ui/core/Dialog';
+import Typography from '@material-ui/core/Typography';
+import { blue } from '@material-ui/core/colors';
+import { Port } from '../hooks/useLogs';
+
+const emails = ['username@gmail.com', 'user02@gmail.com'];
```
Comment (acbeni) [inline react-serial-ui/src/components/ChangePort.tsx:13]: Maybe should be removed?

**Review by acbeni — COMMENTED** [review]
Thanks @yousfiSaad that would make the project much easier to use, just have a few questions and its good to go 👍

File: react-serial-ui/src/components/ChangePort.tsx (hunk 2 above)
Comment (yousfiSaad) [reply react-serial-ui/src/components/ChangePort.tsx:13]: Right

File: react-serial-ui/server/src/index.js (hunk 1 above)
Comment (yousfiSaad) [reply react-serial-ui/server/src/index.js:75]: Yes why not.

**Review by acbeni — APPROVED** [review]
looks all good, thank you very much @yousfiSaad  ❤️

**Comment by yousfiSaad** [comment]
Welcome ya sadi9i

**Pull request merged**